shared/rate_limits/__init__.py

```diff
@@ -39,33 +55,49 @@ def determine_entity_redis_key(
     else:
         auth_info: AdapterAuthInformation = get_adapter_auth_information(owner=owner)
 
-    if auth_info.get("selected_installation_info"):
-        return f"{auth_info.get("selected_installation_info").get("app_id")}_{auth_info.get("selected_installation_info").get("installation_id")}"
-    # How to treat gh app fallbacks?
-    else:
-        return str(auth_info.get("token_owner").ownerid)
+    if (
+        auth_info
+        and auth_info.get("token")
+        and auth_info.get("token").get("entity_name")
+    ):
+        return auth_info.get("token").get("entity_name")
 
```

Comment:
it looks like this function can also return `None`. Please update the return type